portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,44 @@
+function displayComments(){
+    fetch('/comments-data')
+    .catch(error => console.log('failed to fetch comments data from server'))
+    .then(response => response.json())
+    .catch(error => console.log('failed to parse comments'))
+    .then(response => addAllCommentsToDOM(response))
+    .catch(error => console.log('failed to print comments to DOM'));
+}
+
+function createCardHolder(){
+    var card_holder = document.createElement("div");
+    card_holder.setAttribute('class', 'card-holder-small');
+    return card_holder;
+}
+
+function createCard(){
+    var card = document.createElement("div");
+    card.setAttribute('class', 'card-small');
+    return card;
+}
+
+function createParagraph(comment){
+    var para = document.createElement("p"); 
+    para.innerText = comment;
+    return para;
+}
+
+function addCommentToDOM(comment){
+    console.log("displaying comment " + comment)
+    var cardHolder = createCardHolder();
```

Comment:
Nit: here we are using camel-case `cardHolder`, and above we are using underscores to separate words `card_holder`. Either is fine for me. Just pick one and use it consistently throughout. Note that then one you choose might be different in different languages. 
For example in Android we use camel-case for Java, but underscores for xml.